File: JSPM.Web/Modules/Students/Personal/PersonalForm.cs
using Serenity.ComponentModel;
using System;

namespace JSPM.Students.Forms;

[FormScript("Students.Personal")]
[BasedOnRow(typeof(PersonalRow), CheckNames = true)]
public class PersonalForm
{
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string MiddleName { get; set; }
    public string Email { get; set; }
    public int MobileNumber { get; set; }
    public string Pnr { get; set; }
    public string PermanentAddress { get; set; }
    public string City { get; set; }
    public int Pin { get; set; }
    public string CurrentAddress { get; set; }
    public string CurrentCity { get; set; }
    public string Gender { get; set; }
    public DateTime Dob { get; set; }
    public DateTime AdmissionDate { get; set; }
    public int DepartmentId { get; set; }
    public int BranchId { get; set; }
    public int DivisionId { get; set; }
}

File: JSPM.Web/Modules/ServerTypes/Students/PersonalForm.ts
import { StringEditor, IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PersonalForm {
    FirstName: StringEditor;
    LastName: StringEditor;
    MiddleName: StringEditor;
    Email: StringEditor;
    MobileNumber: IntegerEditor;
    Pnr: StringEditor;
    PermanentAddress: StringEditor;
    City: StringEditor;
    Pin: IntegerEditor;
    CurrentAddress: StringEditor;
    CurrentCity: StringEditor;
    Gender: StringEditor;
    Dob: DateEditor;
    AdmissionDate: DateEditor;
    DepartmentId: IntegerEditor;
    BranchId: IntegerEditor;
    DivisionId: IntegerEditor;
}

export class PersonalForm extends PrefixedContext {
    static formKey = 'Students.Personal';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PersonalForm.init)  {
            PersonalForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;

            initFormType(PersonalForm, [
                'FirstName', w0,
                'LastName', w0,
                'MiddleName', w0,
                'Email', w0,
                'MobileNumber', w1,
                'Pnr', w0,
                'PermanentAddress', w0,
                'City', w0,
                'Pin', w1,
                'CurrentAddress', w0,
                'CurrentCity', w0,
                'Gender', w0,
                'Dob', w2,
                'AdmissionDate', w2,
                'DepartmentId', w1,
                'BranchId', w1,
                'DivisionId', w1
            ]);
        }
    }
}